packit/api.py

```diff
@@ -1097,25 +1097,34 @@ def _handle_sources(
 
         # btw this is really naive: the name could be the same but the hash can be different
         # TODO: we should do something when such situation happens
-        archive_name_in_cache = self.dg.is_archive_in_lookaside_cache(
-            self.dg.upstream_archive_name
-        )
+        archives_to_upload = False
         sources_file = self.dg.local_project.working_dir / "sources"
-        archive_name_in_sources_file = (
-            sources_file.is_file()
-            and self.dg.upstream_archive_name in sources_file.read_text()
-        )
+        for upstream_archive_name in self.dg.upstream_archive_names:
+            archive_name_in_cache = self.dg.is_archive_in_lookaside_cache(
+                upstream_archive_name
+            )
+            archive_name_in_sources_file = (
+                sources_file.is_file()
+                and upstream_archive_name in sources_file.read_text()
+            )
 
-        if (
-            archive_name_in_cache
-            and archive_name_in_sources_file
-            and not force_new_sources
-        ):
+            if archive_name_in_cache and archive_name_in_sources_file:
+                continue
+            else:
+                archives_to_upload = True
+        if not archives_to_upload and not force_new_sources:
             return
```

Comment:
With `for-else `, we can get rid of `archives_to_upload` variable:

```suggestion
            if not archive_name_in_cache or not archive_name_in_sources_file:
                break
        else:
            return
```

Reply:
> On the other hand, don't we want to go one by one and do the check for every single one?
> 
> (With your change, we will push all when there is at least one worth the upload.)

That's indeed the intent. The reason is explained in the comment several lines below. Does it make sense?

Reply:
Aha, I missed that... I recall `sources` file update and lookaside-upload can't be done separately.

Will it re-upload the existing source, rise an error or just ignore it?

Reply:
Here's the CLI output.
```
2022-11-15 14:53:42.545 distgit.py        INFO   About to upload to lookaside cache.
2022-11-15 14:53:42.545 commands.py       DEBUG  Command: fedpkg new-sources /tmp/packit-dist-git_3p76ofr/auto-maintenance-49db00dddab301cc55bd4a3708c3f201567f645b.tar.gz /tmp/packit-dist-git_3p76ofr/postfix-1.3.0.tar.gz /tmp/packit-dist-git_3p76ofr/selinux-1.5.0.tar.gz /tmp/packit-dist-git_3p76ofr/timesync-1.7.1.tar.gz /tmp/packit-dist-git_3p76ofr/kdump-1.2.5.tar.gz /tmp/packit-dist-git_3p76ofr/network-1.10.0.tar.gz /tmp/packit-dist-git_3p76ofr/storage-1.9.2.tar.gz /tmp/packit-dist-git_3p76ofr/metrics-1.8.0.tar.gz /tmp/packit-dist-git_3p76ofr/tlog-1.2.9.tar.gz /tmp/packit-dist-git_3p76ofr/kernel_settings-1.1.10.tar.gz /tmp/packit-dist-git_3p76ofr/logging-1.11.0.tar.gz /tmp/packit-dist-git_3p76ofr/nbde_server-1.2.0.tar.gz /tmp/packit-dist-git_3p76ofr/nbde_client-1.2.7.tar.gz /tmp/packit-dist-git_3p76ofr/certificate-1.1.7.tar.gz /tmp/packit-dist-git_3p76ofr/crypto_policies-1.2.6.tar.gz /tmp/packit-dist-git_3p76ofr/ansible-sshd-v0.18.0.tar.gz /tmp/packit-dist-git_3p76ofr/ssh-1.1.11.tar.gz /tmp/packit-dist-git_3p76ofr/ha_cluster-1.8.0.tar.gz /tmp/packit-dist-git_3p76ofr/vpn-1.5.0.tar.gz /tmp/packit-dist-git_3p76ofr/firewall-1.4.0.tar.gz /tmp/packit-dist-git_3p76ofr/cockpit-1.4.0.tar.gz /tmp/packit-dist-git_3p76ofr/podman-1.0.0.tar.gz /tmp/packit-dist-git_3p76ofr/ad_integration-0.0.1.tar.gz /tmp/packit-dist-git_3p76ofr/ci-testing-1.1.1.tar.gz /tmp/packit-dist-git_3p76ofr/ansible-posix-1.4.0.tar.gz /tmp/packit-dist-git_3p76ofr/community-general-6.0.0.tar.gz /tmp/packit-dist-git_3p76ofr/containers-podman-1.9.4.tar.gz
2022-11-15 14:53:46.442 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/auto-maintenance-49db00dddab301cc55bd4a3708c3f201567f645b.tar.gz
2022-11-15 14:53:57.134 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/postfix-1.3.0.tar.gz
2022-11-15 14:53:57.134 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/selinux-1.5.0.tar.gz
2022-11-15 14:53:57.134 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/timesync-1.7.1.tar.gz
2022-11-15 14:53:57.134 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/kdump-1.2.5.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/network-1.10.0.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/storage-1.9.2.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/metrics-1.8.0.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/tlog-1.2.9.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/kernel_settings-1.1.10.tar.gz
2022-11-15 14:53:57.135 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/logging-1.11.0.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/nbde_server-1.2.0.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/nbde_client-1.2.7.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/certificate-1.1.7.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/crypto_policies-1.2.6.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ansible-sshd-v0.18.0.tar.gz
2022-11-15 14:53:57.136 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ssh-1.1.11.tar.gz
2022-11-15 14:53:57.137 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ha_cluster-1.8.0.tar.gz
2022-11-15 14:53:57.137 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/vpn-1.5.0.tar.gz
2022-11-15 14:53:57.188 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/firewall-1.4.0.tar.gz
2022-11-15 14:53:57.709 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/cockpit-1.4.0.tar.gz
2022-11-15 14:53:58.364 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/podman-1.0.0.tar.gz
2022-11-15 14:53:58.881 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ad_integration-0.0.1.tar.gz
2022-11-15 14:53:59.540 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ci-testing-1.1.1.tar.gz
2022-11-15 14:54:00.080 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/ansible-posix-1.4.0.tar.gz
2022-11-15 14:54:00.756 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/community-general-6.0.0.tar.gz
2022-11-15 14:54:01.352 logging.py        INFO   File already uploaded: /tmp/packit-dist-git_3p76ofr/containers-podman-1.9.4.tar.gz
2022-11-15 14:54:01.364 logging.py        INFO   All sources were already uploaded.
```
and the docstring of `upload_to_lookaside_cache` says:
> If the archive is already uploaded, the rpkg tool doesn't do anything.

Reply:
> With `for-else `, we can get rid of `archives_to_upload` variable:

I think this suggestion does not take into account the `not force_new_sources` part.

I think I prefer
```python
            if not archive_name_in_cache or not archive_name_in_sources_file:
                archives_to_upload = True
```
which gets rid of the useless `else:` and `continue`.

Reply:
The last rebase has the simplification I suggested above added.